packages/typespec-ts/test/modularIntegration/dictionaryTypes.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { assert } from "chai";
+import { DictionaryClient } from "./generated/type/dictionary/src/index.js";
+
+describe("DictionaryClient Modular Client", () => {
+  let client: DictionaryClient;
+
+  beforeEach(() => {
+    client = new DictionaryClient({
+      endpoint: "http://localhost:3002",
+      allowInsecureConnection: true,
+      retryOptions: {
+        maxRetries: 0
+      }
+    });
+  });
+
+  it("should get a dictionary of boolean", async () => {
+    try {
+      const result = await client.booleanValue.get();
+      assert.deepEqual(result, { k1: true, k2: false });
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should send a dictionary of boolean", async () => {
+    try {
+      await client.booleanValue.put({ k1: true, k2: false });
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  // Fix deserialization
+  it.skip("should get a dictionary of datetime", async () => {
```

Comment:
I'll revisit this, but this test is giving us the date as string. Probably an issue specific to Record of datetime. I'll see if the fix is scoped I'll fit it in this PR, otherwise I'll address in a follow up

Reply:
This will be addressed in follow up work tracked by: https://github.com/Azure/autorest.typescript/issues/2619